dvc/utils/__init__.py

```diff
@@ -364,15 +366,18 @@ def resolve_paths(repo, out):
     if os.name == "nt" and scheme == abspath.drive[0].lower():
         # urlparse interprets windows drive letters as URL scheme
         scheme = ""
-    if (
-        not scheme
-        and abspath.isin_or_eq(repo.root_dir)
-        and not contains_symlink_up_to(abspath, repo.root_dir)
+
+    if scheme or not abspath.isin_or_eq(repo.root_dir):
+        wdir = os.getcwd()
+    elif contains_symlink_up_to(dirname, repo.root_dir) or (
+        os.path.isdir(abspath) and System.is_symlink(abspath)
     ):
+        raise DvcException(
+            "Cannot add files inside symlinked directories to DVC."
```

Comment:
Looks like we should make it a bit more helpful. Maybe a link to troubleshooting doc?